### PR TITLE
Forbid submission of @temporary-addon guids

### LINKED ISSUE #17579: Deny new add-ons that use an ID ending in @temporary-addon
### Describe the problem and steps to reproduce it:

IDs ending in `@temporary-addon` are reserved for temporarily loaded add-ons in Firefox. We should not accept such IDs for signing (and thus submission).

Since we already have a few add-ons on AMO with that ID, we can't deny any submissions yet, so we are limiting this issue to new add-ons first.

Submitting a new add-on with an ID ending in `@temporary-addon`...

### What happened?

...is accepted by AMO.

### What did you expect to happen?

...should not be accepted by AMO.

### Anything else we should know?

As mentioned above, submission of updates to existing add-ons using such IDs should not be denied for now.

### PR DESCRIPTION
**Pull request by diox**
Fixes #17579

### CODE REVIEW

File: src/olympia/constants/base.py [hunk 1]
```diff
@@ -364,7 +364,7 @@
 
 # Changes to this list need to be synced with the privileged add-ons signing pipeline.
 # Please reach out to the Operations Team (awagner) before making any changes!
```
Comment (eviljeff) [inline src/olympia/constants/base.py:366]: add to this comment as it's not just for privileged add-ons now?

Comment (diox) [reply src/olympia/constants/base.py:366]: Yeah, I wasn't sure exactly what to say here instead. Any suggestion @wagnerand ?

**Review by wagnerand — APPROVED** [review]

File: src/olympia/constants/base.py (hunk 1 above)
Comment (wagnerand) [reply src/olympia/constants/base.py:366]: I think we can say something like

```
# Changes to this list need to be coordinated with the Operations team to make sure other systems like the privileged add-ons signing pipeline are in sync. Please reach out to them before making changes.
```

File: src/olympia/constants/base.py [hunk 2]
```diff
@@ -374,6 +374,7 @@
     '@mozillaonline.com',
     '@mozillafoundation.org',
     '@rally.mozilla.org',
+    '@temporary-addon',
```
Comment (wagnerand) [inline src/olympia/constants/base.py:378]: Maybe also add a specific comment for this line, like
```
# Temporary add-ons as defined in Firefox. They should not be submitted/signed.
```

**Pull request merged**